R/estimateParams.R

```diff
@@ -53,3 +53,366 @@ prepareData <- function(data) {
 
   as.data.frame(dataNew[, -which(names(dataNew) == "time")], row.names = seq_len(nrow(dataNew)))
 }
+
+#' Compute the Negative Log-Likelihood for a Given Data Set and Transition Model
+#'
+#' @param transition (`ExponentialTransition` or `WeibullTransition`)\cr.
+#'   See [exponential_transition()] or [weibull_transition()] for details.
+#' @param data (`data.frame`)\cr
+#'   Data frame in the format created by [prepareData()].
+#'
+#' @return Returns the value of the negative log-likelihood (`numeric`).
+#' @export
+#'
+#' @details
+#' Calculates the negative log-likelihood for a given data set and transition model. It uses the hazard
+#' and survival functions specific to the transition model.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' simData <- getOneClinicalTrial(
+#'   nPat = c(30), transitionByArm = list(transition),
+#'   dropout = list(rate = 0.8, time = 12),
+#'   accrual = list(param = "time", value = 1)
+#' )
+#' negLogLik(transition, prepareData(simData))
+negLogLik <- function(transition, data) {
+  with(data, -sum(log(haz(transition, exit, trans)^status
+    * survTrans(transition, exit, trans) / survTrans(transition, entry, trans))))
+}
+
+#' Hazard Function for Different Transition Models
+#'
+#' @param transition (`ExponentialTransition` or `WeibullTransition`)\cr.
+#'   See [exponential_transition()] or [weibull_transition()] for details.
+#' @param t (`numeric`)\cr time at which hazard is to be computed.
+#' @param trans (`integer`)\cr index specifying the transition type.
+#'
+#' @return Returns the hazard rate (`numeric`) for the specified transition and time.
+#' @export
+#'
+#' @details
+#' This function dispatches to either `haz.ExponentialTransition` or `haz.WeibullTransition`
+#' based on the `transition` object class.
+#'
+#' @examples
+#' transition <- exponential_transition(h01 = 1.2, h02 = 1.5, h12 = 1.6)
+#' haz(transition, 0.4, 2)
+haz <- function(transition, t, trans) {
+  assert_class(transition, "TransitionParameters")
+  assert_numeric(t, lower = 0)
+  assert_subset(trans, c(1, 2, 3))
+  UseMethod("haz")
+}
+
+#' Hazard Function for Exponential Transition Model
+#'
+#' @param transition (`ExponentialTransition`)\cr
+#'   See [exponential_transition()] for details.
+#' @param t (`numeric`)\cr time at which hazard is to be computed.
+#' @param trans (`integer`)\cr index specifying the transition type.
+#'
+#' @return Returns the hazard rate (`numeric`) for the exponential transition at the specified time.
+#' @export
+#'
+#' @details
+#' Computes the hazard function specifically for an exponential transition model using provided parameters.
```

Comment:
This can be shortened to
```suggestion
#' @describeIn haz for an exponential transition model.
#' @export
```